model/src/main/kotlin/utils/OrtResultExtensions.kt

```diff
@@ -26,6 +26,16 @@ import org.ossreviewtoolkit.model.config.PathExclude
 
 import java.util.SortedSet
 
+/**
+ * Return a map of concluded licenses for each package [Identifier] that has a concluded license.
+ */
+fun OrtResult.collectConcludedLicenses(omitExcluded: Boolean = false): Map<Identifier, List<String>> =
+    getPackages(omitExcluded)
+        .filter { it.pkg.concludedLicense != null }
+        .associate {
+            Pair(it.pkg.id, it.pkg.concludedLicense?.licenses().orEmpty())
```

Comment:
You should use `decompose()` instead of `licenses()` in order to handle the `WITH` operator properly.

Reply:
I thought about it again, probably that does not work out as the mapping from ID to text is done separately for licenses and exceptions. Thus the current implementation ignores exceptions. Would it make sense to add these as well to the NOTICE?

Reply:
Exceptions are not handled in the notice report anyway, but I have added a sentence to the docs of the function that makes clear that exceptions are ignored.

Reply:
thx!

Reply:
For re-usability and maybe also the direction these license functions may evolve to, would it make sense to leave the `licenses()` call up to the caller and just return a `Map: Id -> SpdxExpression`? Not sure though due to representing above metnioned "NONE" case.

Reply:
Let's discuss the handling of "NONE" offline and make a follow-up PR if required.

Reply:
My point was more about changing the return type to `Map: Id -> SpdxExpression`, not so much about how to represent NONE. Anyhow, I'll leave it up to you.